crates/unleash-edge/src/edge_builder.rs

```diff
@@ -210,8 +222,213 @@ pub async fn build_edge(
             delta_cache_manager,
             engine_cache,
         ),
-        Arc::new(Some(token_validator)),
-        Arc::new(Some(feature_refresher)),
+        Arc::new(token_validator),
+        Arc::new(feature_refresher),
         persistence,
     ))
 }
+
+pub async fn build_edge_state(
```

Comment:
This is now the primary entry point into the building. Instead of assembling 8 different Arc'd structs, it now just takes what it needs, assembles the AppState and BackgroundTasks and returns those. Moving the background task assembly here means the caller doesn't need to care about whether or not you have a TokenValidator or a MetricsCache etc